DQM/L1TMonitor/src/L1TStage2BMTF.cc

```diff
@@ -35,70 +36,116 @@ void L1TStage2BMTF::beginLuminosityBlock(const edm::LuminosityBlock& iLumi, cons
 
 void L1TStage2BMTF::bookHistograms(DQMStore::IBooker &ibooker, const edm::Run& iRun, const edm::EventSetup& eveSetup)
 {
-  ibooker.setCurrentFolder(monitorDir);
-  
-  bmtf_hwEta = ibooker.book1D("bmtf_hwEta", "HW #eta", 461, -230.5, 230.5);
-  bmtf_hwLocalPhi = ibooker.book1D("bmtf_hwLocalPhi", "HW Local #phi", 201, -100.5, 100.5);
-  bmtf_hwGlobalPhi = ibooker.book1D("bmtf_hwGlobalPhi", "HW Global #phi", 576, -0.5, 575.5);
-  bmtf_hwPt  = ibooker.book1D("bmtf_hwPt", "HW p_{T}", 512, -0.5, 511.5);
-  bmtf_hwQual= ibooker.book1D("bmtf_hwQual" , "HW Quality", 20, -0.5, 19.5);
-  bmtf_proc  = ibooker.book1D("bmtf_proc" , "Processor", 12, -0.5, 11.5);
-
-  bmtf_wedge_bx  = ibooker.book2D("bmtf_wedge_bx", "Wedge vs BX", 12, -0.5, 11.5, 5, -2.5, 2.5);
-  bmtf_wedge_bx->setTitle(";Wedge; BX");
-  for (int bin = 1; bin < 13; ++bin) {
-    bmtf_wedge_bx->setBinLabel(bin, std::to_string(bin), 1);
+  std::string histoPrefix = "bmtf";
+  if (kalman) {
+    histoPrefix = "kbmtf";
   }
 
-  bmtf_hwEta_hwLocalPhi = ibooker.book2D("bmtf_hwEta_hwLocalPhi", "HW #eta vs HW Local #phi" , 461, -230.5, 230.5, 201, -100.5, 100.5);
-  bmtf_hwEta_hwLocalPhi->setTitle(";HW #eta; HW Local #phi");
+  ibooker.setCurrentFolder(monitorDir);
+  if (!kalman) {
```

Comment:
Remove the if and delete the else block. Then put only the additional histograms for Kalman in an `if (kalman)`.